server/src/main/java/org/elasticsearch/search/aggregations/bucket/terms/SignificantTextAggregatorFactory.java

```diff
@@ -253,13 +259,15 @@ private CollectorSource createCollectorSource() {
             MappedFieldType fieldType,
             Analyzer analyzer,
             String[] sourceFieldNames,
```

Comment:
I wonder if we could flatmap the sourceFieldNames in the code that calls the ctor.